test/index.js

```diff
@@ -170,22 +171,138 @@ describe('@apostrophecms/import-export', function () {
           title: 'topic3',
           aposMode: 'draft'
         },
+        {
+          title: 'topic3',
+          aposMode: 'published'
+        },
+        {
+          title: 'topic1',
+          aposMode: 'published'
+        },
         {
           title: 'topic2',
           aposMode: 'draft'
         },
         {
-          title: 'topic1',
+          title: 'topic2',
           aposMode: 'published'
+        }
+      ],
+      attachmentsLength: 1,
+      attachmentFiles: [ `${attachmentId}-test-image.jpg` ],
+      topicsContainNonProjectedFields: true
+    };
+
+    assert.deepEqual(actual, expected);
+  });
+
+  it('should generate a zip file for pieces with related documents in all modes', async function () {
+    const req = apos.task.getReq();
+    const { _id: attachmentId } = await apos.attachment.db.findOne({ name: 'test-image' });
+    const manager = apos.article;
+
+    // PREPARE
+    // update an article draft removing the topic relation
+    await apos.doc.db.updateOne(
+      {
+        title: 'article1',
+        aposMode: 'draft'
+      },
+      { $set: { topicsIds: [] } }
+    );
+    const articles1 = await manager.find(req, {
+      title: 'article1',
+      aposMode: { $in: [ 'draft', 'published' ] }
+    }).locale(null).toArray();
+    {
+      const draft = articles1.find(({ aposMode }) => aposMode === 'draft');
+      const published = articles1.find(({ aposMode }) => aposMode === 'published');
+      assert.deepEqual(draft._topics, []);
+      assert.equal(published._topics.length, 1);
+    }
```

Comment:
omg we can scope constants this way, simply with curly brackets?
this will change my life

Reply:
I can't decide how I feel about it but it's cool heh